### PR TITLE
CLC Oxidation Property Updates

### PR DESCRIPTION
**Pull request by bpaul4**
## Fixes
- Addresses currently failing BFB iron oxidation flowsheet example by applying unit model scaling, and adding thermophysical/reaction property scaling and associated tests. Note that the bulk of the "additions" are duplicating the BFB test module to enable coverage of both property packages.

## Changes proposed in this PR:
- Expands iron oxidation properties for new scaling, unit definitions and improvements
- Adds scaling and unit tests for gas, solid and reaction properties
- Renaming existing BFB test module and adding a second module to cover both methane reduction and iron oxidation
- Minor other fixes to iron oxidation, methane reduction and CLC unit model files

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Review by chineduobiora — APPROVED** [review]
The changes look good.

File: idaes/models_extra/gas_solid_contactors/properties/oxygen_iron_OC_oxidation/gas_phase_thermo.py [hunk 1]
```diff
@@ -823,11 +995,17 @@ def cp_mol(b):
 
     def _cp_mass(self):
         # Mixture heat capacities
+        units_meta = self._params.get_metadata().derived_units
+        units_cp_mass = (
```
Comment (andrewlee94) [inline idaes/models_extra/gas_solid_contactors/properties/oxygen_iron_OC_oxidation/gas_phase_thermo.py:999]: `units_meta["heat_capacity_mass"]`

File: idaes/models_extra/gas_solid_contactors/properties/oxygen_iron_OC_oxidation/gas_phase_thermo.py [hunk 2]
```diff
@@ -844,23 +1022,31 @@ def cp_mass(b):
 
     def _enth_mol_comp(self):
         # Pure component vapour enthalpies
+        units_meta = self._params.get_metadata().derived_units
+        units_enth_mol = units_meta["energy"] * units_meta["amount"] ** -1
```
Comment (andrewlee94) [inline idaes/models_extra/gas_solid_contactors/properties/oxygen_iron_OC_oxidation/gas_phase_thermo.py:1026]: `units_meta["energy_mole"]`

File: idaes/models_extra/gas_solid_contactors/properties/oxygen_iron_OC_oxidation/gas_phase_thermo.py [hunk 3]
```diff
@@ -899,17 +1087,105 @@ def _enth_mol(self):
             self.del_component(self.mixture_enthalpy_eqn)
             raise
 
-    def get_material_flow_terms(b, p, j):
-        return b.flow_mol * b.mole_frac_comp[j]
+    def _entr_mol(self):
+        units_meta = self._params.get_metadata().derived_units
+        units_entr_mol = (units_meta["energy"] * units_meta["amount"] ** -1
```
Comment (andrewlee94) [inline idaes/models_extra/gas_solid_contactors/properties/oxygen_iron_OC_oxidation/gas_phase_thermo.py:1092]: `units_meta["entropy_mole"]`

File: idaes/models_extra/gas_solid_contactors/properties/oxygen_iron_OC_oxidation/hetero_reactions.py [hunk 4]
```diff
@@ -432,17 +442,21 @@ def _k_rxn(self):
             self._params.rate_reaction_idx,
             domain=Reals,
             initialize=1,
-            doc="Rate constant " "[mol^(1-N_reaction)m^(3*N_reaction -2)/s]",
+            doc="Rate constant " "[mol^(1-rxn_order) * m^(3*rxn_order -2)/s]",
```
Comment (andrewlee94) [inline idaes/models_extra/gas_solid_contactors/properties/oxygen_iron_OC_oxidation/hetero_reactions.py:445]: Extra quotation marks in middle of doc string,

File: idaes/models_extra/gas_solid_contactors/properties/oxygen_iron_OC_oxidation/gas_phase_thermo.py [hunk 5]
```diff
@@ -115,13 +107,13 @@ def build(self):
         )
 
         # Std. heat of formation of comp. - units = kJ/(mol comp) - ref: NIST
-        enth_mol_form_comp_dict = {"O2": 0, "N2": 0, "CO2": -393.5224, "H2O": -241.8264}
+        enth_mol_form_comp_dict = {"O2": 0e3, "N2": 0e3, "CO2": -393.5224e3, "H2O": -241.8264e3}
```
Comment (andrewlee94) [inline idaes/models_extra/gas_solid_contactors/properties/oxygen_iron_OC_oxidation/gas_phase_thermo.py:110]: Minor comment, but `0e3` look a bit odd. I would just do `0`.

File: idaes/models_extra/gas_solid_contactors/properties/oxygen_iron_OC_oxidation/solid_phase_thermo.py [hunk 6]
```diff
@@ -229,10 +302,10 @@ def define_metadata(cls, obj):
                     "method": "_dens_mass_particle",
                     "units": "kg/m3",
                 },
-                "cp_mol_comp": {"method": "_cp_mol_comp", "units": "kJ/mol.K"},
-                "cp_mass": {"method": "_cp_mass", "units": "kJ/kg.K"},
-                "enth_mass": {"method": "_enth_mass", "units": "kJ/kg"},
-                "enth_mol_comp": {"method": "_enth_mol_comp", "units": "kJ/mol"},
+                "cp_mol_comp": {"method": "_cp_mol_comp", "units": "J/mol.K"},
```
Comment (andrewlee94) [inline idaes/models_extra/gas_solid_contactors/properties/oxygen_iron_OC_oxidation/solid_phase_thermo.py:305]: Minor comment for reference - the "units" part is now redundant and not used, so you do not need to include it.

File: idaes/models_extra/gas_solid_contactors/properties/oxygen_iron_OC_oxidation/solid_phase_thermo.py [hunk 7]
```diff
@@ -413,49 +486,54 @@ def build(self):
...
-            units=pyunits.kg / pyunits.s,
+            doc="Component mass flowrate",
+            units=units_meta["mass"] / units_meta["time"],
         )
         self.particle_porosity = Var(
             domain=Reals,
             initialize=0.27,
             doc="Porosity of oxygen carrier [-]",
-            units=pyunits.m**3 / pyunits.m**3,
+            units=units_meta["length"] ** 3 / units_meta["length"] ** 3,
```
Comment (andrewlee94) [inline idaes/models_extra/gas_solid_contactors/properties/oxygen_iron_OC_oxidation/solid_phase_thermo.py:503]: Due to the way `pint` works, you might as well just make this `pyunits.dimensionless`.

File: idaes/models_extra/gas_solid_contactors/properties/oxygen_iron_OC_oxidation/solid_phase_thermo.py [hunk 8]
```diff
@@ -413,49 +486,54 @@ def build(self):
...
         )
         self.mass_frac_comp = Var(
             self._params.component_list,
             initialize=1 / len(self._params.component_list),
             doc="State component mass fractions [-]",
-            units=pyunits.kg / pyunits.kg,
+            units=units_meta["mass"] / units_meta["mass"],
```
Comment (andrewlee94) [inline idaes/models_extra/gas_solid_contactors/properties/oxygen_iron_OC_oxidation/solid_phase_thermo.py:509]: As above.

File: idaes/models_extra/gas_solid_contactors/properties/oxygen_iron_OC_oxidation/solid_phase_thermo.py [hunk 9]
```diff
@@ -506,21 +585,31 @@ def density_particle_constraint(b):
 
     def _cp_mol_comp(self):
         # Pure component solid heat capacities
+        units_meta = self._params.get_metadata().derived_units
+        units_cp_mol = (
```
Comment (andrewlee94) [inline idaes/models_extra/gas_solid_contactors/properties/oxygen_iron_OC_oxidation/solid_phase_thermo.py:589]: As before, you can get this directly from `unit_meta`.

See line 249 in https://github.com/IDAES/idaes-pse/blob/main/idaes/core/base/property_meta.py for reference about which units are already defined (and if something is missing, you should add it there).

**Review by andrewlee94 — COMMENTED** [review]
A few fairly minor things to fix - mostly about getting derived units from `units_meta`. Otherwise, looks good.

File: idaes/models_extra/gas_solid_contactors/properties/oxygen_iron_OC_oxidation/solid_phase_thermo.py (hunk 9 above)
Comment (bpaul4) [reply idaes/models_extra/gas_solid_contactors/properties/oxygen_iron_OC_oxidation/solid_phase_thermo.py:589]: Thanks for looking at this and the other units tags, I'll cross-reference `property_meta.py` and reduce to derived units where I can.

File: idaes/models_extra/gas_solid_contactors/properties/oxygen_iron_OC_oxidation/solid_phase_thermo.py (hunk 6 above)
Comment (bpaul4) [reply idaes/models_extra/gas_solid_contactors/properties/oxygen_iron_OC_oxidation/solid_phase_thermo.py:305]: Would you recommend removing "units" on all metadata lines, for the gas/solid/reaction modules?

Comment (andrewlee94) [reply idaes/models_extra/gas_solid_contactors/properties/oxygen_iron_OC_oxidation/solid_phase_thermo.py]: It doesn't hurt things if it is there, but it never actually gets used either.

**Review by chineduobiora — APPROVED** [review]

**Pull request merged**